plutarch-context-builder/src/Plutarch/Context/Transformer.hs

```diff
@@ -0,0 +1,375 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Plutarch.Context.Transformer (
+  -- * Data type
+  Transformer,
+  -- * Introduction
+  -- ** Generic
+  liftDatumEndo,
+  liftDatumKleisli,
+  liftDatumCheck,
+  liftRedeemerEndo,
+  liftRedeemerKleisli,
+  liftRedeemerCheck,
+  liftScriptContextEndo,
+  liftScriptContextKleisli,
+  liftScriptContextCheck,
+  liftValidatorEndo,
+  liftValidatorKleisli,
+  liftValidatorCheck,
+  liftMintingPolicyEndo,
+  liftMintingPolicyKleisli,
+  liftMintingPolicyCheck,
+  -- ** Specific checkers
+  txInfoRedeemersAreSorted,
+  txInfoOutputsAreSorted,
+  -- ** Specific normalizers
+  sortTxInfoRedeemers,
+  sortTxInfoOutputs,
+  -- * Elimination
+  runValidatorTransformer,
+  runValidatorTransformerPure,
+  runMintingPolicyTransformer,
+  runMintingPolicyTransformerPure
+  ) where
+
+import Control.Applicative ((<|>))
+import Data.Void (Void, absurd)
+import Control.Monad.Except (MonadError (throwError))
+import Data.Bifunctor (first)
+import Control.Arrow (Kleisli (Kleisli))
+import Data.Kind (Type)
+import Control.Category ((>>>), id)
+import Prelude hiding (id)
+import Data.Profunctor (Star (Star))
+import PlutusLedgerApi.V2 (ScriptContext (scriptContextTxInfo), 
+  txInfoRedeemers, ScriptPurpose (Minting, Spending, Rewarding, Certifying), 
+  txInfoOutputs, txOutReferenceScript)
+import Data.Profunctor.Rep (tabulate)
+import Data.Monoid (Endo (Endo, appEndo))
+import PlutusTx.AssocMap qualified as AssocMap
+import Data.List (sort, sortOn)
+
+-- | @since 3.0.1
+newtype Transformer (d :: Type) (r :: Type) (err :: Type) = 
+  Transformer (Star (Either err) (Maybe d, r, ScriptContext) (Maybe d, r, ScriptContext))
```

Comment:
To requests for documentation here:
- "Transformer" seems to be a pretty overloaded term. We should specify what this is.
- "Star" is not a type I think many people are familiar with. We're not exposing the internals, but I think a non-haddock comment as to what this similar to